src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+                                     for translation, darkOffset, digitalGain in zip(cellTranslation,celldarkOffset,celldigitalGain)],
+                    x_scan_to_acq_delay=int(self.acqDelay.value - self.parent._EBeamScanner.scanDelay.value[0]),
+                    x_scan_delay=self.parent._EBeamScanner.scanDelay.value[0],
+                    y_prescan_lines=self.parent._EBeamScanner.scanDelay.value[1],
+                    flyback_time=int(self.parent._EBeamScanner.scanFlyback.value /
+                                   self.parent._EBeamScanner.clockPeriod.value),
+                    x_scan_offset=self.parent._EBeamScanner.scanOffset.value[0],
+                    y_scan_offset=self.parent._EBeamScanner.scanOffset.value[1],
+                    x_scan_gain=self.parent._EBeamScanner.scanGain.value[0],
+                    y_scan_gain=self.parent._EBeamScanner.scanGain.value[1],
+                    x_descan_gain=self.parent._MirrorDescanner.scanGain.value[0],
+                    y_descan_gain=self.parent._MirrorDescanner.scanGain.value[1],
+                    x_descan_offset=self.parent._MirrorDescanner.scanOffset.value[0],
+                    y_descan_offset=self.parent._MirrorDescanner.scanOffset.value[1],
+                    scan_rotation=self.parent._EBeamScanner.rotation.value,
+                    descan_rotation=self.parent._MirrorDescanner.rotation.value,
+            )
+
+        return self._megafield_metadata
+
+    def _acquire(self):
+        try:
+            self._current_status = None
+            tnext = 0
+
+            while True:
+                # Wait until a message is available
+                command, *meta_info = self.acq_queue.get(block=True)
+                if command == "start" and self._current_status != "start":
+                    self._current_status = command
+                    self.parent.simpleASMCall(self._server_URL + "/scan/start_mega_field", _METHOD_POST, 204, meta_info[
+                        0].to_dict())
+
+                elif command == "next" and self._current_status != "stop":
+                    self.parent.simpleASMCall(self._server_URL + "/scan/scan_field", _METHOD_POST, 204,
+                                              meta_info[0].to_dict())
+
+                    # TODO add functionality for getting a full and thumbnail image if simulator is updated
+                    if meta_info[1] != 'empty':
+                        logging.error("Option %s not yet implemented" % meta_info[1])
+                        img = self.parent.simpleASMCall(self._server_URL + "/scan/field", _METHOD_GET, 200, (meta_info[0].position_x,
+                                                                                                             meta_info[0].position_y, self._data_content2value(meta_info[1]).to_dict()))
+                        self.data.notify(model.DataArray(img, dtype=numpy.uint8), self._getMetadata())
```

Comment:
Here I had following in mind:
```suggestion
                    img_content = self._data_content2value(dataContent.value)
                     if img_content is None:
                           self.data.notify(model.DataArray(<empty array>])
                     else:
                          img = self.parent.simpleASMCall(<......, thumbnail = img_content>, ....)
                           self.data.notify(<img>)
```
However, I agree, if the VA changed before this piece of code was executed, we have a problem. So indeed, it might make more sense to put "next" with "dataContent.value" info on the queue. @pieleric Do you see any problems here or can suggest an elegant way to prevent this issue with the VA changing its value and the need to put more info on the queue...

Reply:
I did split it up, as Sabrina suggested boolean is enough for this. Though I also used None(instead of False) as output for termination which is handy for the test cases)